presto-native-execution/presto_cpp/main/types/PrestoToVeloxQueryPlan.cpp

```diff
@@ -1996,18 +1996,76 @@ VeloxQueryPlanConverterBase::toVeloxQueryPlan(
   auto insertTableHandle =
       std::make_shared<core::InsertTableHandle>(connectorId, hiveTableHandle);
 
-  auto outputType = toRowType(
-      {node->rowCountVariable,
-       node->fragmentVariable,
-       node->tableCommitContextVariable});
+  std::vector<protocol::VariableReferenceExpression> variables = {
+      node->rowCountVariable,
+      node->fragmentVariable,
+      node->tableCommitContextVariable};
+
+  std::shared_ptr<core::AggregationNode> aggregationNode;
```

Comment:
Can you provide a utility say node = buildStatsAggregationNode(protocol node, &variables)?